### PR TITLE
Fix sort to work correctly for numbers

### PR DESCRIPTION
**Pull request by roll**
- connects https://github.com/BCODMO/frictionless-usecases/issues/21

### CODE REVIEW

**Comment by roll** [comment]
@akariv 
@cschloer
This one is more like a request for a discussion than a PR... Because this issue - https://github.com/BCODMO/frictionless-usecases/issues/21 - is kinda tricky. As I see `sort_rows` is fully string-based processors comparison-wise.

Consider it as POC for handling number and date/times but I'm not sure about performance and maybe there is a better way to achieve the same goal (e.g. mutating the `key` instead of every value).

WDYT?

**Comment by roll** [comment]
The initial plan was too naive. Let's see what we can do else..

File: dataflows/processors/sort_rows.py [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
+import datetime
 from kvfile import KVFile
-
 from ..helpers.resource_matcher import ResourceMatcher
 
 
 class KeyCalc(object):
     def __init__(self, key_spec):
         self.key_spec = key_spec
+        self.key_list = re.findall(r'\{(.*?)\}', key_spec)
```
Comment (akariv) [inline dataflows/processors/sort_rows.py]: I think that a better implementation would be to use a custom Formatter, and add a new 'float type' (or override the existing one) that produces the reversed notation:
https://docs.python.org/3.7/library/string.html#string.Formatter.format_field

wdyt?

Comment (akariv) [reply dataflows/processors/sort_rows.py]: @roll

Comment (roll) [reply dataflows/processors/sort_rows.py]: good idea. let me check

**Comment by roll** [comment]
@akariv 
It seems I made it work but it looks very complicated and will have some limitations (mistakes in comparing big numbers etc)

TBH I'm not sure what's direction is the most promising...

I thought about different approaches using real number comparison and all of them seems to be much less effective memory or performance-wise. We probably can use a `KVFile` modification with a numeric key but it's a limited case for one column ordering.

WDYT?

**Comment by roll** [comment]
Here is an update from @akariv I'm going to use to update the PR:

> If we use the binary representation of the float, it will ‘just work’
> 
> https://www.h-schmidt.net/FloatConverter/IEEE754.html
> and no need for expensive formatting / transformations
> just dump the float, as it’s in memory, to a hex string
> 
> https://stackoverflow.com/questions/16444726/binary-representation-of-float-in-python-bits-not-hex

**Comment by roll** [comment]
@akariv 
Please take a look

The new solution based on binary arrays seems much nicer, passes the same tests and hopefully more performant and precise (for cases like 100000000000001).

**Comment by akariv** [comment]
Looks good @roll! 

Do the tests cover the Decimal case as well?

**Comment by roll** [comment]
@akariv 
I've added the real data file sorting which also covers decimals sorting

**Pull request merged**